render/gles2/renderer.c

```diff
@@ -120,16 +120,36 @@ static bool gles2_render_texture_with_matrix(struct wlr_renderer *wlr_renderer,
 
 	GLuint prog = 0;
 	GLenum target = 0;
+	GLint proj_loc = 0;
+	GLint invert_y_loc = 0;
+	GLint tex_loc = 0;
+	GLint alpha_loc = 0;
+
 	switch (texture->type) {
 	case WLR_GLES2_TEXTURE_GLTEX:
 	case WLR_GLES2_TEXTURE_WL_DRM_GL:
-		prog = texture->has_alpha ? renderer->shaders.tex_rgba :
-			renderer->shaders.tex_rgbx;
+		if (texture->has_alpha) {
+			prog = renderer->shaders.tex_rgba.program;
```

Comment:
Using the same struct for these 3 would allow a more concise code:

```c
struct wlr_gles2_tex_shader *shader = NULL;

if (…) {
  shader = &renderer->shaders.tex_rgba;
}
…
```